### DISCUSSION TITLE
试图把WSL从C盘迁移到D盘，失败了，现在无法使用WSL

### DISCUSSION THREAD

**Discussion by CresDai** [physics-data/faq#175] [Q&A] [answered]
由于C盘空间将满，借鉴了网上的方法（<https://blog.csdn.net/u014175785/article/details/118181230>）试图将WSL移动到D盘。我使用了如下命令：
```
> wsl --export Ubuntu D:\Ubuntu_bak.tar
> wsl --unregister Ubuntu
> wsl --import Ubuntu_new D:\WSL_Ubuntu D:\Ubuntu_bak.tar --version 2
> ubuntu config --default-user cresdai
```
随后我的wsl分发状态和进入wsl环境后的状态变成了这样：
```
PS C:\Users\Crescent_Dai> wsl -l -v
  NAME          STATE           VERSION
* Ubuntu_new    Running         2
  Ubuntu        Running         2
PS C:\Users\Crescent_Dai> wsl
root@LAPTOP-QVK2GDML:/mnt/c/Users/Crescent_Dai# code
```
通过该命令进入code后，发现之前的工作区和文件均无法打开，提示“资源不可用”或"... does not exist"。
请问我是不是少了某些操作，导致wsl无法找到我的文件位置？

**Comment by Harry-Chen** [comment, marked as answer]
注意你的最后一条命令 `ubuntu config --default-user cresdai` 作用在了错误的发行版上。你应该对于新的 `Ubuntu_new` 设置默认用户，才能获得预期的效果。

你可以修改 `/etc/wsl.conf`，添加：

```text
[user]
default=cresdai
```

然后重启 WSL。

Reference: <https://superuser.com/questions/1566022/how-to-set-default-user-for-manually-installed-wsl-distro>.
